packages/transformers/js/src/JSTransformer.js

```diff
@@ -768,17 +769,60 @@ export default (new Transformer({
       asset.meta.hasCJSExports = hoist_result.has_cjs_exports;
       asset.meta.staticExports = hoist_result.static_cjs_exports;
       asset.meta.shouldWrap = hoist_result.should_wrap;
-    } else if (needs_esm_helpers) {
-      asset.addDependency({
-        specifier: '@parcel/transformer-js/src/esmodule-helpers.js',
-        specifierType: 'esm',
-        resolveFrom: __filename,
-        env: {
-          includeNodeModules: {
-            '@parcel/transformer-js': true,
+    } else {
+      if (symbol_result) {
```

Comment:
Is the following duplicated? Should we combine `hoist_result` and `symbol_result` somehow?